### PR TITLE
Change Phone to StudentID and change Person to Student

### LINKED ISSUE #86: Change Phone attribute to StudentID, and refactor Person to Student
Make it more convenient and appropriate to our FYPManager theme

### PR DESCRIPTION
**Pull request by eugenelmj**
- Change the Phone attribute to StudentID for the whole code
- Update the UG to reflect the correct input of StudentID using prefix "id/'
- Change "Person" to "Student" to suit our JerylFYP manager theme better
- Check thoroughly to ensure the above are changed and reflected correctly.

This update will make the presentation of the students' particulars more aligned to our desired outcome for v1.2. 

This issue resolves #86.

### CODE REVIEW

File: src/main/java/jeryl/fyp/commons/core/Messages.java [hunk 1]
```diff
@@ -7,7 +7,7 @@ public class Messages {
 
     public static final String MESSAGE_UNKNOWN_COMMAND = "Unknown command";
     public static final String MESSAGE_INVALID_COMMAND_FORMAT = "Invalid command format! \n%1$s";
-    public static final String MESSAGE_INVALID_PERSON_DISPLAYED_INDEX = "The person index provided is invalid";
+    public static final String MESSAGE_INVALID_STUDENT_DISPLAYED_INDEX = "The person index provided is invalid";
```
Comment (RussellDash332) [inline src/main/java/jeryl/fyp/commons/core/Messages.java:10]: Should this be 'student index'?

File: src/main/java/jeryl/fyp/model/student/exceptions/DuplicateStudentException.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package jeryl.fyp.model.student.exceptions;
+
+/**
+ * Signals that the operation will result in duplicate Students (Students are considered duplicates if they have the
+ * same identity).
+ */
+public class DuplicateStudentException extends RuntimeException {
+    public DuplicateStudentException() {
+        super("Operation would result in duplicate persons");
```
Comment (RussellDash332) [inline src/main/java/jeryl/fyp/model/student/exceptions/DuplicateStudentException.java:9]: 'duplicate students' to sync up the rest

**Review by RussellDash332 — CHANGES_REQUESTED** [review]
I'll leave out some non-code parts such as LICENSE and tutorial markdown files as it's not really required for this change.

Some small changes required are commented below. Also, do you think #88 can be included here as well?

File: src/main/java/jeryl/fyp/commons/core/Messages.java (hunk 1 above)
Comment (eugenelmj) [reply src/main/java/jeryl/fyp/commons/core/Messages.java:10]: You mean to just put as `MESSAGE_INVALID_STUDENT_INDEX`?

File: src/main/java/jeryl/fyp/model/student/exceptions/DuplicateStudentException.java (hunk 2 above)
Comment (eugenelmj) [reply src/main/java/jeryl/fyp/model/student/exceptions/DuplicateStudentException.java:9]: Good catch! Fixed.

**Comment by eugenelmj** [comment]
Yup! I think #88 fits here as well. Should we resolve that?

**Comment by RussellDash332** [comment]
> Yup! I think #88 fits here as well. Should we resolve that?

Sure!

File: src/main/java/jeryl/fyp/commons/core/Messages.java (hunk 1 above)
Comment (RussellDash332) [reply src/main/java/jeryl/fyp/commons/core/Messages.java:10]: As in the string itself still uses 'person index'

Comment (eugenelmj) [reply src/main/java/jeryl/fyp/commons/core/Messages.java:10]: Ah! Silly me. Thanks for that! Fixed.

**Comment by eugenelmj** [comment]
This issue resolves #88.

**Comment by RussellDash332** [comment]
> This issue resolves #88.
> 
> Update: Yeah, seems like we have no choice but to resolve it separately.

No worries, so it doesn't yet as of now 👍🏻

**Review by RussellDash332 — APPROVED** [review]

**Pull request merged**